crates/wit/src/lib.rs

```diff
@@ -273,6 +274,56 @@ struct PublishOptions<'a> {
     dry_run: bool,
 }
 
+fn add_registry_metadata(config: &Config, bytes: &[u8]) -> Result<Vec<u8>> {
+    let mut metadata = RegistryMetadata::default();
+    if !config.authors.is_empty() {
+        metadata.set_authors(Some(config.authors.clone()));
+    }
+
+    if !config.categories.is_empty() {
+        metadata.set_categories(Some(config.categories.clone()));
+    }
+
+    metadata.set_description(config.description.clone());
+
+    // TODO: registry metadata should have keywords
+    // if !package.keywords.is_empty() {
```

Comment:
Are `keywords` distinct from `categories`? They seem coupled in the original registry discussion.

Reply:
`crates.io` has them as [distinct fields](https://doc.rust-lang.org/cargo/reference/manifest.html#the-keywords-field), where keywords is for describing the package and the category with a set of predefined [slugs](https://crates.io/category_slugs) to use.

I wasn't following the discussion around the metadata implementation closely, so it sounds like we should be populating this off of `keywords` instead of categories or perhaps both?

Reply:
I'm going to switch to populating the `categories` field from the cargo `keywords` as I believe that's more of the intention of this field; the predefined categories list from cargo doesn't map to what we'd want in a component registry anyway.

I've opened https://github.com/bytecodealliance/registry/issues/197 to track the confusion here.

Reply:
I've decided to merge this as-is and wait for the resolution on the linked issue; will update `cargo-component` if things change.